test/react/hooks/use-chat-client.test.tsx

```diff
@@ -19,10 +20,15 @@ vi.mock('ably');
 
 describe('useChatClient', () => {
   it('should throw an error if used outside of ChatClientProvider', () => {
-    expect(() => render(<TestComponent callback={() => {}} />)).toThrowErrorInfo({
-      code: 40000,
-      message: 'useChatClient hook must be used within a chat client provider',
-    });
+    const TestThrowError: React.FC = () => {
+      expect(() => useRoom()).toThrowErrorInfo({
```

Comment:
Asking to learn. How are we sure in this test that this `expect()` is called?

Reply:
So I'm fairly sure the operation up until this point is synchronous :)